esmvaltool/diag_scripts/kcs/resampling.py

```diff
@@ -127,50 +132,74 @@ def select_final_subset(combinations, n=8):
+
+    Step 1a: Get 1000 combinations for the control period.
+    These samples should have the same mean winter precipitation as the
+    overall mean of the x ensemble members.
+
+    Step 1b: Get 1000 combinations for the future period.
+    The target value is a relative change wrt,
+    the overall mean of the control period.
+
+    Step 2a
+    Step 2b
+    Step 3
+    """
 
-    # Step 0: Read the data, extract segmented subsets for both the control
-    # and future periods, and precompute seasonal means for each segment.
+    # Start step 0:
+    # Read the data
+    # TODO define a function to get input data
     dataset_dicts = cfg['input_data'].values()
     target_model_metadata = select_metadata(
-        dataset_dicts, dataset=cfg['target_model'])
+        dataset_dicts, dataset=cfg['target_model']
+    )
     files = [metadata['filename'] for metadata in target_model_metadata]
-    dataset = xr.open_mfdataset(
-        files, concat_dim='ensemble_member', combine='nested')[['pr', 'tas']]
-    segments_season_means = {}
+
+    # TODO nor working, it needs another dimension for variables!
+    dataset = xr.open_mfdataset(files, concat_dim='ensemble_member', combine='nested')[['pr', 'tas']]
+    n_ensemble_members = len(dataset.ensemble_member)
 
     # Precompute the segment season means for the control period ...
-    segments = segment(dataset, *cfg['control_period'], step=5)
+    segments_season_means = {}
+    segments = segment(dataset, cfg['control_period'], step=5)
     season_means = segments.groupby('time.season').mean()
     segments_season_means['control'] = season_means
 
     # ... and for all the future scenarios
     for name, info, in cfg['scenarios']:
-        segments = segment(dataset, *info['resampling_period'], step=5)
+        segments = segment(dataset, info['resampling_period'], step=5)
         season_means = segments.groupby('time.season').mean()
         segments_season_means[name] = season_means
 
+    # Start step 1:
     # Create a dictionary to store the selected indices later on.
     selected_indices = {}
 
     # Step 1a: Get 1000 combinations for the control period
-    # These samples should have the same mean winter precipitation as the
-    # overall mean of the x ensemble members
+    # Get mean winter precipitation per segment
     winter_mean_pr_segments = segments_season_means['control'].pr.sel(
         season='DJF')
+    # Get overall mean of segments
     target = winter_mean_pr_segments.mean()
-    top1000 = most_promising_combinations(winter_mean_pr_segments, target)
-    selected_indices['control'] = top1000
+    # Select top1000 values
+    selected_indices['control'] = most_promising_combinations(
+        winter_mean_pr_segments, target, n_ensemble_members
+    )
 
     # Step 1b: Get 1000 combinations for the future period
-    # The target value is a relative change wrt the overall mean of the control period
+    # The target value is the overall mean of the control period
```

Comment:
This was correct before: "The target value is **a relative change wrt** the overall mean of the control period". We're looking at the future period in step 1b.